### PR TITLE
Misc adjustments to dependency tracking

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/internal/utils/dependency_tracking.py [hunk 1]
```diff
@@ -4,10 +4,10 @@
     "FileDependenciesStrategy",
 ]
 
+import logging
```
Comment (Raalsky) [inline src/neptune/internal/utils/dependency_tracking.py:7]: We have our custom logger somewhere. It has some settings to simplify filtering etc. I think it's:
```python
from neptune.internal.utils.logger import logger
```

File: src/neptune/internal/utils/dependency_tracking.py [hunk 2]
```diff
@@ -32,7 +32,9 @@ def log_dependencies(self, run: "Run") -> None:
 class InferDependenciesStrategy(DependencyTrackingStrategy):
     def log_dependencies(self, run: "Run") -> None:
         try:
-            dependencies_str = subprocess.check_output([sys.executable, "-m", "pip", "freeze"]).decode("utf-8")
+            dependencies_str = subprocess.check_output([sys.executable, "-m", "pip", "list", "--format=freeze"]).decode(
```
Comment (Raalsky) [inline src/neptune/internal/utils/dependency_tracking.py:35]: Have you tried the `importlib` approach? I've been suggesting something like:
```python
import sys

if sys.version_info >= (3, 8):
    from importlib.metadata import distributions
else:
    from importlib_metadata import distributions


def list_installed_packages():
    dists = list(sorted(distributions(), key=lambda d: d.metadata['Name']))

    for dist in dists:
        name, version = dist.metadata['Name'], dist.metadata['Version']
        print(f'{name}=={version}')


def main():
    list_installed_packages()


if __name__ == '__main__':
    main()
```

Comment (AleksanderWWW) [reply src/neptune/internal/utils/dependency_tracking.py:35]: @SiddhantSadangi

File: src/neptune/internal/utils/dependency_tracking.py (hunk 1 above)
Comment (AleksanderWWW) [reply src/neptune/internal/utils/dependency_tracking.py:7]: I tried that at first and it doesn't show the level - just the message. I could change the logging format, but that would affect other usages of this logger throughout the code base, right?

File: src/neptune/internal/utils/dependency_tracking.py (hunk 2 above)
Comment (SiddhantSadangi) [reply src/neptune/internal/utils/dependency_tracking.py:35]: Just tried. This seems to work very well!

Lets do this instead of subprocess

**Review by Raalsky — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,11 @@
-## [UNRELEASED] neptune 1.3.2
+##  [UNRELEASED] neptune 1.3.3
+
+### Changes
+- Dependency tracking feature will log an error if a given file path doesn't exist ([#1389](https://github.com/neptune-ai/neptune-client/pull/1389))
+- Use `pip list --format=freeze` instead of `pip freeze` in dependency tracking ([#1389](https://github.com/neptune-ai/neptune-client/pull/1389))
```
Comment (twolodzko) [inline CHANGELOG.md:5]: Not sure if it was fixed, but this note is misleading as we're not using it here.

**Pull request merged**